### PR TITLE
Implementing the `EXPLAIN` Parser

### PR DESCRIPTION
**Pull request by iifawzi**
Hi, This PR is adding and supporting the EXPLAIN Statements with all its variations for MariaDB and MySQL.
Following up on https://github.com/phpmyadmin/phpmyadmin/issues/17482#issuecomment-1312471693, and resolving https://github.com/phpmyadmin/phpmyadmin/issues/11867.

Signed-off-by: iifawzi <iifawzie@gmail.com>

### CODE REVIEW

File: tests/Builder/ExplainStatementTest.php [hunk 1]
```diff
@@ -9,15 +9,59 @@
...
 
+        /* Assertion 6 */
+        $query = 'DESC FOR CONNECTION 458';
+        $parser = new Parser($query);
+        $stmt = $parser->statements[0];
         $this->assertEquals(
-            ' EXPLAIN SELECT * FROM test',
+            'DESC FOR CONNECTION 458',
             $stmt->build()
         );
```
Comment (iifawzi) [inline tests/Builder/ExplainStatementTest.php:66]: This gives an overview of the different variations this PR should support, hopefully, I'm not missing any case

**Review by williamdes — APPROVED** [review]
Looks really good

**Comment by williamdes** [comment]
Thank you for your continuous work and contributions on this parser ! 🚀

**Comment by williamdes** [comment]
Fixed a typo -> c4d58c0245dfac3a4dc6e05ee8f34752e573129c

**Pull request merged**